aidial_adapter_vertexai/utils/resource.py

```diff
@@ -1,58 +1,54 @@
 import base64
-import mimetypes
 import re
-from pathlib import Path
-from typing import Optional, Self
+from typing import Optional
 
-from aidial_sdk.chat_completion import Attachment
 from pydantic import BaseModel
-from vertexai.preview.generative_models import Part
 
 
 class Resource(BaseModel):
-    """
-    Resource: byte value + MIME type
-    """
-
-    mime_type: str
+    type: str
     data: bytes
 
     @classmethod
-    def from_path(cls, path: Path) -> Self:
-        mime_type = mimetypes.guess_type(path)[0]
-        if mime_type is None:
-            raise ValueError(f"Could not determine MIME type for {path}")
-        return cls(mime_type=mime_type, data=path.read_bytes())
+    def from_base64(cls, type: str, data_base64: str) -> "Resource":
+        try:
+            data = base64.b64decode(data_base64, validate=True)
+        except Exception:
+            raise ValueError("Invalid base64 data")
+
+        return cls(type=type, data=data)
 
     @classmethod
-    def from_data_url(cls, url: str) -> Optional[Self]:
+    def from_data_url(cls, data_url: str) -> Optional["Resource"]:
         """
+        Parsing a resource encoded as a data URL.
         See https://developer.mozilla.org/en-US/docs/Web/HTTP/Basics_of_HTTP/Data_URLs for reference.
         """
 
-        pattern = r"^data:([^;]+);base64,(.+)$"
-        match = re.match(pattern, url)
-        if match is None:
+        type = cls.parse_data_url_content_type(data_url)
+        if type is None:
             return None
 
-        mime_type = match.group(1)
-        base64_data = match.group(2)
+        data_base64 = data_url.removeprefix(cls._to_data_url_prefix(type))
 
-        try:
-            data = base64.b64decode(base64_data, validate=True)
-        except Exception:
-            raise ValueError("Invalid base64 data")
-
-        return cls(mime_type=mime_type, data=data)
+        return cls.from_base64(type, data_base64)
 
-    def get_base64_data(self) -> str:
+    @property
+    def data_base64(self) -> str:
         return base64.b64encode(self.data).decode()
 
-    def to_part(self) -> Part:
-        return Part.from_data(data=self.data, mime_type=self.mime_type)
+    def to_data_url(self) -> str:
+        return f"{self._to_data_url_prefix(self.type)}{self.data_base64}"
+
+    @staticmethod
+    def parse_data_url_content_type(data_url: str) -> Optional[str]:
```

Comment:
Why did you decide to remove previous

```python
mimetypes.guess_type
```
 approach?

Reply:
`mimetypes.guess_type` would have returned the content type, but it wouldn't have ensured that the URL is **data URL**, which is what `parse_data_url_content_type` method also does.